### ISSUE TITLE
Granting a role to a user wipes all direct privileges

### ISSUE THREAD

**Issue by proddata** [crate/crate#16710]
### CrateDB version

5.8.3

### CrateDB setup information

Single node local

### Problem description

The privileges granted directly to a user should take precedence over those inherited from a role.
After granting a role to a user who already has direct ALL privileges, querying the sys.users table results in a `SchemaUnknownException`. This behavior is unexpected because direct privileges should take precedence over role-based privileges.

### Steps to Reproduce

**crate**
```
~/ crash
cr> CREATE USER my_user WITH (password='my_password');
cr> GRANT ALL TO my_user;

```

**my_user**
```
~/ crash -U my_user -W
cr> SELECT current_user;
+--------------+
| CURRENT_USER |
+--------------+
| my_user      |
+--------------+
cr> SELECT * FROM sys.users;
+---------------+------+---------+----------+-----------+
| granted_roles | jwt  | name    | password | superuser |
+---------------+------+---------+----------+-----------+
| []            | NULL | crate   | NULL     | TRUE      |
| []            | NULL | my_user | ******** | FALSE     |
+---------------+------+---------+----------+-----------+
```

**crate**
```
cr> CREATE ROLE my_role2;
cr> GRANT my_role2 TO my_user;
```

**my_user**
```
cr> SELECT * FROM sys.users;
SchemaUnknownException[Schema 'sys' unknown]
```

### Actual Result

After assigning the role my_role2 to my_user, attempting to query sys.users results in a SchemaUnknownException.

### Expected Result

The user my_user should continue to have access to the sys.users table since they have direct ALL privileges.

**Comment by proddata** [comment]
On a 2nd look, it seems like this is actually wipeing the privileges granted initially:

```
cr> SELECT * FROM sys.privileges limit 100;
+--------+---------+---------+-------+-------+------+
| class  | grantee | grantor | ident | state | type |
+--------+---------+---------+-------+-------+------+
| SCHEMA | my_role | crate   | x     | GRANT | DQL  |
+--------+---------+---------+-------+-------+------+

cr> SELECT * FROM sys.users limit 100;
+--------------------------------------------+------+---------+----------+-----------+
| granted_roles                              | jwt  | name    | password | superuser |
+--------------------------------------------+------+---------+----------+-----------+
| []                                         | NULL | crate   | NULL     | TRUE      |
| [{"grantor": "crate", "role": "my_role2"}] | NULL | my_user | ******** | FALSE     |
+--------------------------------------------+------+---------+----------+-----------+
SELECT 2 rows in set (0.006 sec)
```

**Comment by seut** [comment]
@proddata Thank you for reporting. We've fixed this by https://github.com/crate/crate/pull/16713, the fix will be released with the next `5.8.4` hotfix release.
